src/fragments/Slider.react.js

```diff
@@ -72,15 +92,57 @@ export default class Slider extends Component {
               )
             : this.props.marks;
 
+        const computedStyle = this._computeStyle(
+            vertical,
+            verticalHeight,
+            tooltip
+        );
+
+        const defaultInputStyle = {
+            width: '60px',
+            marginRight: vertical && syncedInput ? '' : '25px',
+            marginBottom: vertical && syncedInput ? '25px' : '',
+        };
+
         return (
             <div
                 id={id}
                 data-dash-is-loading={
                     (loading_state && loading_state.is_loading) || undefined
                 }
                 className={className}
-                style={this._computeStyle(vertical, verticalHeight, tooltip)}
+                style={{...computedStyle, ...style}}
             >
+                {syncedInput ? (
+                    <this.SyncedInput
+                        onChange={event => {
+                            event.persist();
+                            if (this.timeout) {
+                                clearTimeout(this.timeout);
+                            }
+                            this.timeout = setTimeout(
+                                function() {
+                                    this.syncInput(event);
+                                }.bind(this),
+                                syncedInputDebounceTime
+                            );
+                        }}
+                        onBlur={event => {
+                            this.syncInput(event);
+                        }}
+                        onKeyPress={event => {
+                            if (event.key === 'Enter') {
+                                this.syncInput();
```

Comment:
We should probably clear the timeout here and in `onBlur` (or just inside `syncInput`?) so we don't call `syncInput` twice in these cases.

Reply:
wait, this particular call has no `event` so does that mean `syncInput` won't do anything?

Reply:
This comment is was correct, but has been made outdated by refactoring this component to avoid needing to pass an `event` around at all. I did take your suggestion and clear the timeout in the `syncInputWithSlider` function to ensure it is always cleared no matter which event handler is invoked.